More Winter Coats - Expansion Pack

Nakalon opened the pull request:
## About The Pull Request

This PR adds more Winter Coats to the game.
Because Winter Coats are cute, dressing up is fun, and some of them needed a touch up.

The following were added:
![image (23)](https://github.com/BeeStation/BeeStation-Hornet/assets/94647521/f9f8b318-f1f9-4a29-8b6f-3b8392a8caef)
1 - Virologist
2 - Geneticist
3 - Chemist
4 - Cargo
5 - Mining
6 - Brig Physician
7 - Detective

Within medbay, Viro, Geneticist and chemist, I took the armor values of the medbay coat and allocated them in a way that makes sense to me. They have colours relating to their job. They are available in their respective vendors.

Virologist has more more bio resistance, less of acid.
Geneticist has less bio and more rad. (The geneticist design has the features of a medcoat and the stripe of a science coat, because genetics is like a science)
Chemist has no bio but has Fire and a bit of bomb.
Chemist winter coat is also flame and acid-resistant. (This is not really a balance choice, more of a flavor one. These only really affect the item)

Cargo's Coat had its sprite modernized. It was painted in the colours of the old coat.
The Old Cargo coat can be bough on the hacked cargo vendor.
Cargo's Coat now accepts items that are relevant to the job of cargo (like the export scanner, clipboard, and drink bottles because cargo are hardy men?)

Mining Coat had its sprite modernized. It resembles the exploration suit miners use.
The Old coat was already mostly unobtainable and since its ugly I just kept it that way.
This was made to have similar functionality to the exploration suit, it is meant to be used in real mining work.
It has almost the same values but less bio and rad protection.
It can be upgraded with goliath plates all the same, and it can store miner related items in its slot:
Pickaxes, PKA, Resonator, Scanner, and again, a DRINK BOTTLE.
It is available in the miner equipment lockers.

The Brig Physician Coat bears a resemblance to the vest.
It has less armor values than the vest, but a tiny bit of laser to account for friendly fire incidents.
Its available on the brig phys locker.

The Detective Coat was also made to resemble the detective's jacket.
It has less armour values than the jacket, overall.
It fits all the items important for Detective activities such as the Peperspray, scanner, wooden baton, gun, etc.

## Why It's Good For The Game

This gives a tiny bit of love to some jobs that don't get many new things to cherish.
I can't exactly explain how much I like winter coats, and I love the colours of viro and chemist specially, so I really thought this would add a good amount of colour and choice for players.

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>

![image (23)](https://github.com/BeeStation/BeeStation-Hornet/assets/94647521/0bbf9828-e05b-4389-8b42-823abbc6fb69)
![image (24)](https://github.com/BeeStation/BeeStation-Hornet/assets/94647521/06ba6e7a-4e55-4988-8b72-1f2e6723c6e9)
![image (25)](https://github.com/BeeStation/BeeStation-Hornet/assets/94647521/a24397bd-6db1-4d46-880a-7509cae58570)
![image (26)](https://github.com/BeeStation/BeeStation-Hornet/assets/94647521/dfd354dd-e8e4-4d8f-a70b-7369df8e0bc9)
![image (27)](https://github.com/BeeStation/BeeStation-Hornet/assets/94647521/f9246943-5061-444d-834b-c67a710e8719)
![image](https://github.com/BeeStation/BeeStation-Hornet/assets/94647521/da2fce04-0288-4f43-ac87-d26eac133e3a)


</details>

## Changelog
:cl: PinkSuzuky
add: Added 5 Winter Coats, Virologist, Geneticist, Chemist, Brig Physician and Detective. Made them available on the relevant vendors, with the exception of the Brig Physician Coat which can be acquired from the Brig Physician Locker.
imageadd: Updated the sprites of the Miner Winter Coat and Cargo Winter coat.
tweak: Changed some values on the Miner Winter Coat to make usable for mining.
tweak: Made the Miner Winter Coat available inside the mining equipment lockers.
tweak: Cargo Winter Coat now accepts Cargo related items in its item slot.
code: reorganized some lists on wintercoat code for readeability.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Nakalon committed:
Adds More wintercoats

Real Epicly

Nakalon committed:
quick fix my bad

sloppy fingers

Tsar-Salat commented on code/modules/clothing/suits/miscellaneous.dm:
```diff
@@ -634,6 +641,57 @@
+		/obj/item/gun/ballistic/revolver/detective,
+		/obj/item/detective_scanner,
+		/obj/item/flashlight,
+		/obj/item/taperecorder
```
```suggestion
		/obj/item/taperecorder,
```

Nakalon replied:
guh my bad, I'll hold on on fixing this, I thinking I misinterpreted how the lists on subtypes would interact with the original ones.
Are they additive or do they replace completely the parent list?

Tsar-Salat replied:
Adding a list to the subtype redefines it. It will not inherit the parents list if you redefine it.

Nakalon replied:
Then I'll have to redo it tomorrow ACH, thanks...

Nakalon committed:
forgot a comma

woopie

Co-authored-by: Tsar-Salat <62388554+Tsar-Salat@users.noreply.github.com>

Nakalon committed:
Readded allowed objects form parent

I thougth I was being smart by removing the objects from the allowed items common to the parent coat but i was actually being dumb.
This fixes it.

Nakalon committed:
Merge branch 'wintercoatexpansion' of https://github.com/Nakalon/BeeStation-Hornet into wintercoatexpansion

itsmeow commented on code/modules/clothing/suits/miscellaneous.dm:
```diff
@@ -713,20 +898,71 @@
+	. = ..()
+	AddComponent(/datum/component/armor_plate)
+
+/obj/item/clothing/head/hooded/explorer/Initialize(mapload)
```
is this meant to be for the miner coat?

Nakalon replied:
It was! I coded this part at 4 am, guh, its fixed now!!!

Nakalon committed:
fixes explorer/mining blunder
